### PR TITLE
Handle when AccessToken is disposed

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

Handle when AccessToken is disposed

## PR Context

Customers are reporting that they are getting an error that this is already disposed when we try to dispose it. 
I don't find anywhere else we dispose it, so, I'm catching the error.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs [hunk 1]
```diff
@@ -2745,9 +2746,15 @@ public void Dispose()
                     // If we initialized with thread impersonation, make sure de-initialize is run with the same.
                     if (_identityToImpersonate != null)
                     {
```
Comment (TravisEz13) [inline src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs:2748]:
```suggestion
                    if (_identityToImpersonate != null) {
```

File: src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs [hunk 2]
```diff
@@ -2743,11 +2744,16 @@ public void Dispose()
 
 #if !UNIX
                     // If we initialized with thread impersonation, make sure de-initialize is run with the same.
-                    if (_identityToImpersonate != null)
-                    {
+                    if (_identityToImpersonate != null) {
```
Comment (TravisEz13) [inline src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs:2747]:
```suggestion
                    if (_identityToImpersonate != null) 
                    {
```

File: src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs [hunk 3]
```diff
@@ -2743,11 +2744,16 @@ public void Dispose()
...
+                        try {
```
Comment (TravisEz13) [inline src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs:2748]:
```suggestion
                        try 
                        {
```

File: src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs [hunk 4]
```diff
@@ -2743,11 +2744,18 @@ public void Dispose()
 
 #if !UNIX
                     // If we initialized with thread impersonation, make sure de-initialize is run with the same.
-                    if (_identityToImpersonate != null)
+                    if (_identityToImpersonate != null)
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs:2747]:
```suggestion
                    if (_identityToImpersonate != null)
```

File: src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs [hunk 5]
```diff
@@ -2743,11 +2744,18 @@ public void Dispose()
...
-                    if (_identityToImpersonate != null)
+                    if (_identityToImpersonate != null) 
                     {
+                        try 
+                        {
                         result = WindowsIdentity.RunImpersonated(
                             _identityToImpersonate.AccessToken,
                             () => WSManNativeApi.WSManDeinitialize(_handle, 0));
+                        }
+                        catch (ObjectDisposedException ex)
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs:2755]: This seems like a patch, do we know why the object is already disposed?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Pull request closed without merging**